### ISSUE TITLE
MMM-NesteBussAtB: No buses

### ISSUE THREAD

**Issue by padlefot** [ottopaulsen/MMM-NesteBussAtB#7]
Hei!

Det står bare "loading" - Jeg ser i konsollen at den kaster: MMM-NesteBussAtB: No buses - 
MMM-NesteBussAtB.js:98 

Undrer på om det kan være en form for geo sperre da jeg laster fra en Engelsk (UK) IP?

Noen tips?

Her er config:

`{
    module: 'MMM-NesteBussAtB',
    position: 'top_bar',
    header: "Bus Departures & Traffic Conditions in Trondheim",
    config: {
        showIcon: true, // Bus icon in front of row
        showNumber: true, // Bus number
        showFrom: true, // Bus stop name
        showTo: true, // Bus destination
        showMin: true, // "min" text after minutes
        size: "small", // Text size, for example small, medium or large
        stopIds: [16010001, 16010002, 16010004, 16010005], // See https://www.atb.no/holdeplassoversikt/
        maxCount: 2, // Max number of next buses per route
        maxMinutes: 45, // Do not show buses more then this minutes into the future
        stacked: true // Show multiple buses on same row, if same route and destination       
    }
},`

**Comment by ottopaulsen** [comment]
Hei

Jeg vet ikke om noen slik sperre.
Jeg har sett at det av og til står loading. Det skjer f.eks. hvis det ikke er noen busser, men med de holdeplassene du har satt opp er det jo busser hele dagen (Munkegata). Kanskje du prøvde om natta? 
Ellers har jeg ingen forslag annet enn å prøve igjen.
Du kan jo også se i loggen om der vises noen form for feilmelding.

Otto

**Comment by padlefot** [comment]
Takk for kjapt svar :) Og ikke minst en veldig kul modul!

Ja det er en underlig sak,

Loggen viser ingenting spesielt:
`MMM-NesteBussAtB: Received socket notification NESTEBUSSATB_CONFIG
MMM-NesteBussAtB: AtB Connection started`

Jeg prøvde både i går og i dag under arbeidstid (09-16) men får det samme. Det ser ut som konsollen hamrer requester mot Atb for jeg får "MMM-NesteBussAtB: No buses -
MMM-NesteBussAtB.js:98" ca hvert andre sekund og konsollen fyller seg opp med dem. 

Skal prøve å debugge dette litt dypere men sitter litt fast for øyeblikket, alle tips mottas med takk!

Det funker hjemme hos meg la det være sagt så jeg mistenker fortsatt at noe er endret her jeg er på jobb mht nettverk eller WAN.

Mvh
padle

**Comment by ottopaulsen** [comment]
Ja, den sender en del requests, for den poller AtB hvert halvminutt for alle stoppene. Det skal vist være mulig å sette opp til å bruke callback i stedet for polling, men jeg har ikke funnet noen dokumentasjon på API-et, og heller ikke brukt så mye tid for å finne ut av det.

DOM-en oppdateres ganske ofte, så det kommer vel No buses hver gang. Det kan sikkert gjøres bedre :-)

**Comment by ottopaulsen** [comment]
Har du kommet videre med denne?

Otto

**Comment by padlefot** [comment]
Hey! Nei jeg har ikke rukket å se mer på det desverre, har samme problemet enda dog og det må nok være noe innenfor nettverket her på jobb (fw, etc), da det funker hjemmefra.

**Comment by padlefot** [comment]
Jeg ser at de vanlige terminalene fra MinSkjerm "loader" urimelig treigt her og; Eksempel er: [AtB MinSkjerm](http://st.atb.no/New/minskjerm/FST.aspx?visMode=2&cTit=Bus%20Departures%20-%20ARM%20Norway%20/%20Munkegata&c1=1&s1=16010001&sv1=&cn1=Munkegata%20M1&c2=1&s2=16010002&sv2=&cn2=Munkegata%20M2&c3=1&s3=16010004&sv3=&cn3=Munkegata%20M4&c4=1&s4=16010005&sv4=&cn4=Munkegata%20M5&template=1&cmhb=33CCCC&cmhc=000000&cshb=000000&cshc=FFFFFF&arb=000000&rows=20&period=&) - Noen ganger, spesielt i iframes, loader de ikke i det hele tatt og jeg får en "Feil på internettforbindelsen fra dette brukerstedet" error fra AtB.
